product/stream_processor/integrations/events/providers/eventbridge.py

```diff
@@ -0,0 +1,87 @@
+import os
+from typing import TYPE_CHECKING, Generator, Optional
+
+import boto3
+import botocore.exceptions
+
+from product.constants import XRAY_TRACE_ID_ENV
+from product.stream_processor.integrations.events.base import EventProvider
+from product.stream_processor.integrations.events.constants import EVENTBRIDGE_PROVIDER_MAX_EVENTS_ENTRY
+from product.stream_processor.integrations.events.exceptions import ProductNotificationDeliveryError
+from product.stream_processor.integrations.events.functions import chunk_from_list
+from product.stream_processor.integrations.events.models.input import Event
+from product.stream_processor.integrations.events.models.output import EventReceipt, EventReceiptFail, EventReceiptSuccess
+
+if TYPE_CHECKING:
+    from mypy_boto3_events import EventBridgeClient
+    from mypy_boto3_events.type_defs import PutEventsRequestEntryTypeDef, PutEventsResponseTypeDef
+
+
+class EventBridge(EventProvider):
```

Comment:
in the producrization pr we will add logs/metrics/traces right?

Reply:
Spot on! Left that in the PR body so changes are scoped and easier to review. Parser and data validation in depth will come there too.